segment_terraform.go

```diff
@@ -0,0 +1,24 @@
+package main
+
+type terraform struct {
+	props         *properties
+	env           environmentInfo
+	workspaceName string
+}
+
+func (tf *terraform) string() string {
+	return tf.workspaceName
+}
+
+func (tf *terraform) init(props *properties, env environmentInfo) {
+	tf.props = props
+	tf.env = env
+}
+
+func (tf *terraform) enabled() bool {
+	if !(tf.env.hasCommand("terraform") && tf.env.hasFolder(".terraform")) {
```

Comment:
Can we adjust this to `!tf.env.hasCommand("terraform") && !tf.env.hasFolder(".terraform")`, when I read this I didn't see the initial `!` and I'm sure I won't be the only one :-)

Reply:
Actually, it seems like if !(A && B) and  if !A && !B does not evaluate in the same manner.

With the first implementation (how it was) it did not show the prompt if there was not a .terraform directory, but in the suggested implementation it actually shows the prompt despite not having a .terraform directory. I've tested it with the current build.

Is this a go-ism? @JanDeDobbeleer  Agree that for readability the suggested implementation is better.

Reply:
A suggestion for how to include this test into segment_terraform_test.go (actually evaluate both cases together) would be of great help! :)

Reply:
It should be `!tf.env.hasCommand("terraform") || !tf.env.hasFolder(".terraform")`, when either of one condition isn't met, it's disabled.